### ISSUE TITLE
"No peer for topic skipping publish" / "Attestation sent" inconsistency

### ISSUE THREAD

**Issue by mratsim** [status-im/nim-beacon-chain#3269]
From discord:

```
eth2_1        | INF 2022-01-08 11:20:47.000+00:00 Slot start                                 topics="beacnde" slot=2901402 epoch=90668 sync=synced peers=122 head=0fa49ad1:2901401 finalized=90666:6d42cb5e delay=771us948ns
eth2_1        | NOT 2022-01-08 11:20:50.151+00:00 No peers for topic, skipping publish       topics="libp2p gossipsub" peersOnTopic=0 connectedPeers=0 topic=/eth2/afcaaba0/beacon_attestation_45/ssz_snappy
eth2_1        | NOT 2022-01-08 11:20:50.151+00:00 Attestation sent                           topics="beacval" attestation="(aggregation_bits: 0b00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000, data: (slot: 2901402, index: 45, beacon_block_root: \"c50ceb66\", source: \"90667:27813a1d\", target: \"90668:3524a785\"), signature: \"<REDACTED>\")" validator=<REDACTED> delay=-848ms601us579ns subnet_id=45
eth2_1        | NOT 2022-01-08 11:20:50.155+00:00 Attestation sent                           topics="beacval" attestation="(aggregation_bits: 0b0000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, data: (slot: 2901402, index: 48, beacon_block_root: \"c50ceb66\", source: \"90667:27813a1d\", target: \"90668:3524a785\"), signature: \"<REDACTED>\")" validator=<REDACTED> delay=-844ms273us234ns subnet_id=48
eth2_1        | NOT 2022-01-08 11:20:55.008+00:00 Aggregated attestation sent                topics="beacval" attestation="(aggregation_bits: 0b00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000, data: (slot: 2901402, index: 45, beacon_block_root: \"c50ceb66\", source: \"90667:27813a1d\", target: \"90668:3524a785\"), signature: \"<REDACTED>\")" aggregator_index=<REDACTED> signature=<REDACTED> validator=<REDACTED> subnet_id=45
eth2_1        | INF 2022-01-08 11:20:55.410+00:00 Slot end                                   topics="beacnde" slot=2901402 nextActionWait=3s589ms773us642ns nextAttestationSlot=2901403 nextProposalSlot=-1 head=c50ceb66:2901402
```

The logs give the impression that an attestation is sent, but there is no check that the subnet actually has peers:

https://github.com/status-im/nimbus-eth2/blob/67ab477a27e358d605e99bffeb67f98d18218eca/beacon_chain/validators/validator_duties.nim#L392-L394

https://github.com/status-im/nimbus-eth2/blob/67ab477a27e358d605e99bffeb67f98d18218eca/beacon_chain/validators/validator_duties.nim#L980-L985

https://github.com/status-im/nimbus-eth2/blob/67ab477a27e358d605e99bffeb67f98d18218eca/beacon_chain/validators/validator_duties.nim#L1182-L1183

Also while having no peers might be worth a NOTICE log level in libp2p (which doesn't have enough context), 95% of the work of a validator is attesting so such an error should be a warning at the very least or maybe an error in Nimbus.

**Comment by Menduist** [comment]
`pubsub.publish` will return to how many peers we broadcasted to, so you could use that to log correctly (eg, 0 = error, < something = warning, else OK)

**Comment by zah** [comment]
A fix for this has been merged in the `unstable` branch and will be shipped in Nimbus v22.6.0
